econml/tests/test_inference.py

```diff
@@ -370,6 +371,18 @@ def test_auto_inference(self):
         est.marginal_effect_inference(T, X).summary_frame()
         assert est.marginal_effect_inference(T, X).stderr is not None
 
+    def test_pickle_inferenceresult(self):
+        Y, T, X, W = TestInference.Y, TestInference.T, TestInference.X, TestInference.W
+        est = DML(model_y=LinearRegression(),
+                  model_t=LinearRegression(),
+                  model_final=Lasso(alpha=0.1, fit_intercept=False),
+                  featurizer=PolynomialFeatures(degree=1, include_bias=False),
+                  random_state=123)
+        est.fit(Y, T, X=X, W=W)
+        effect_inf = est.effect_inference(X)
+        with open("test.pkl", "wb") as file:
+            pickle.dump(effect_inf, file)
```

Comment:
You can just dump to a string instead

```suggestion
        s = pickle.dumps(effect_inf)
```